src/SonarScanner.MSBuild.Shim/AdditionalFilesService.cs

```diff
@@ -80,8 +86,14 @@ public AdditionalFiles AdditionalFiles(AnalysisConfig analysisConfig, DirectoryI
             .EnumerateFiles(projectBaseDir.FullName, "*", SearchOption.AllDirectories)
             .Select(x => new FileInfo(x))
             .Where(x => extensions.Any(e => x.Name.EndsWith(e, StringComparison.OrdinalIgnoreCase) && !x.Name.Equals(e, StringComparison.OrdinalIgnoreCase)))
+            .Where(x => !ContainsExcludedDirectory(x))
             .ToList();
 
+    private static bool ContainsExcludedDirectory(FileInfo fileInfo) =>
+        Array.Exists(
+            fileInfo.DirectoryName.Split(Path.DirectorySeparatorChar),
+            x => ExcludedDirectories.Exists(excluded => x.Equals(excluded, StringComparison.OrdinalIgnoreCase)));
```

Comment:
This is a lot of nested loops. It is more efficient to do this in `GetAllFiles`:
```cs
    private List<FileInfo> GetAllFiles(IEnumerable<string> extensions, DirectoryInfo projectBaseDir) =>
        directoryWrapper
            .EnumerateDirectories(projectBaseDir.FullName, "*", SearchOption.AllDirectories)
            .Where(x => !ExcludedDirectories.Exists(..))
            .SelectMany(x => x.EnumerateFiles("*", SearchOption.TopDirectoryOnly)
            .Where(x => extensions.Any(e => x.Name.EndsWith(e, StringComparison.OrdinalIgnoreCase) && !x.Name.Equals(e, StringComparison.OrdinalIgnoreCase)))
            .ToList();
```